core/src/main/java/org/geysermc/geyser/translator/protocol/java/JavaLoginTranslator.java

```diff
@@ -108,6 +109,13 @@ public void translate(GeyserSession session, ClientboundLoginPacket packet) {
         gamerulePacket.getGameRules().add(new GameRuleData<>("doimmediaterespawn", !packet.isEnableRespawnScreen()));
         session.sendUpstreamPacket(gamerulePacket);
 
+        // Prevent issues such as https://github.com/GeyserMC/Geyser/issues/5489
```

Comment:
Maybe add an extra line here?

Resend the death info if the player is already dead on login